packages/web/app/forms/PatientDetailsForm/layouts/generic/patientFields/GenericLocationFields.jsx

```diff
@@ -20,40 +27,51 @@ export const GenericLocationFields = ({ filterByMandatory }) => {
         <TranslatedText stringId="general.localisedField.cityTown.label" fallback="City/town" />
       ),
     },
-    subdivisionId: {
-      component: AutocompleteField,
-      suggester: subdivisionSuggester,
-      label: (
-        <TranslatedText
-          stringId="general.localisedField.subdivisionId.label"
-          fallback="Sub division"
-        />
-      ),
-    },
-    divisionId: {
-      component: AutocompleteField,
-      suggester: divisionSuggester,
-      label: (
-        <TranslatedText stringId="general.localisedField.divisionId.label" fallback="Division" />
-      ),
-    },
+    ...(isUsingLocationHierarchy
+      ? {}
+      : {
+          subdivisionId: {
+            component: AutocompleteField,
+            suggester: subdivisionSuggester,
+            label: (
+              <TranslatedText
+                stringId="general.localisedField.subdivisionId.label"
+                fallback="Sub division"
+              />
+            ),
+          },
+          divisionId: {
+            component: AutocompleteField,
+            suggester: divisionSuggester,
+            label: (
+              <TranslatedText
+                stringId="general.localisedField.divisionId.label"
+                fallback="Division"
+              />
+            ),
+          },
+        }),
     countryId: {
       component: AutocompleteField,
       suggester: countrySuggester,
       label: (
         <TranslatedText stringId="general.localisedField.countryId.label" fallback="Country" />
       ),
     },
-    settlementId: {
-      component: AutocompleteField,
-      suggester: settlementSuggester,
-      label: (
-        <TranslatedText
-          stringId="general.localisedField.settlementId.label"
-          fallback="Settlement"
-        />
-      ),
-    },
+    ...(isUsingLocationHierarchy
```

Comment:
```suggestion
    ...(isUsingLocationHierarchy && {
          settlementId: {
            component: AutocompleteField,
            suggester: settlementSuggester,
            label: (
              <TranslatedText
                stringId="general.localisedField.settlementId.label"
                fallback="Settlement"
              />
            ),
          },
        })),
```
would work in these cases